src/apply.c

```diff
@@ -413,12 +415,21 @@ int git_apply__patch(
 			newfile->mode : GIT_FILEMODE_BLOB;
 	}
 
-	if (patch->delta->flags & GIT_DIFF_FLAG_BINARY)
-		error = apply_binary(contents_out, source, source_len, patch);
+	if (patch->delta->flags & GIT_DIFF_FLAG_BINARY) {
+		if (!(ctx.opts.flags & GIT_APPLY_CHECK)) {
+			error = apply_binary(contents_out, source, source_len, patch);
+		} else {
+			// TODO
```

Comment:
Sorry for this nit after such a long time, but we're trying to be as C89 conformant as possible. So this should be a `/* */` style comment, but I'd be happy enough to just remove the comment altogether

Reply:
Curious - are there changes needed for `apply_binary` to support `GIT_APPLY_CHECK`?  I would have expected not.  Was the omission just because you don't need support for it or are there actually changes needed there?